src/components/Api.js

```diff
@@ -0,0 +1,79 @@
+export default class Api {
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  async _handleApiResponses(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  async _apiRequest(url, options) {
```

Comment:
Function names should start with a **verb** in infinitive (there is such a point in the check list). It’s common in coding to do it so that functions could `speak` for themselves about what they do.


`_makeRequest` or just `_request`